homeassistant/components/smappee/binary_sensor.py

```diff
@@ -0,0 +1,173 @@
+
+    @property
+    def unique_id(self,):
+        """Return the unique ID for this binary sensor."""
+        return (
+            f"{self._service_location.device_serial_number}-"
+            f"{self._service_location.service_location_id}-"
+            f"presence"
+        )
+
+    @property
+    def device_info(self):
+        """Return the device info for this binary sensor."""
+        return {
+            "identifiers": {(DOMAIN, self._service_location.device_serial_number)},
+            "name": self._service_location.service_location_name,
+            "manufacturer": "Smappee",
+            "model": self._service_location.device_model,
+            "sw_version": self._service_location.firmware_version,
+        }
+
+    async def async_update(self):
+        """Get the latest data from Smappee and update the state."""
+        await self._smappee_base.async_update()
+
+        self._state = self._service_location.is_present
+
+
+class SmappeeAppliance(BinarySensorEntity):
+    """Implementation of a Smappee binary sensor."""
+
+    def __init__(
+        self,
+        smappee_base,
+        service_location,
+        appliance_id,
+        appliance_name,
+        appliance_type,
+    ):
+        """Initialize the Smappee sensor."""
+        self._smappee_base = smappee_base
+        self._service_location = service_location
+        self._appliance_id = appliance_id
+        self._appliance_name = appliance_name
+        self._appliance_type = appliance_type
+        self._state = False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return (
+            f"{self._service_location.service_location_name} - "
+            f"{BINARY_SENSOR_PREFIX} - "
+            f"{self._appliance_name if self._appliance_name != '' else self._appliance_type}"
+        )
+
+    @property
+    def is_on(self):
+        """Return if the binary sensor is turned on."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        icon_mapping = {
+            "Car Charger": "mdi:car",
+            "Coffeemaker": "mdi:coffee",
+            "Clothes Dryer": "mdi:tumble-dryer",
+            "Clothes Iron": "mdi:hanger",
+            "Dishwasher": "mdi:dishwasher",
+            "Lights": "mdi:lightbulb",
+            "Fan": "mdi:fan",
+            "Freezer": "mdi:fridge",
+            "Microwave": "mdi:microwave",
+            "Oven": "mdi:stove",
+            "Refrigerator": "mdi:fridge",
+            "Stove": "mdi:stove",
+            "Washing Machine": "mdi:washing-machine",
+            "Water Pump": "mdi:water-pump",
+        }
+        return icon_mapping.get(self._appliance_type)
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        # Only lights can be mapped onto the generic list of binary sensors
+        return "light" if self._appliance_type == "Lights" else "power"
```

Comment:
When we set a device class to something other than `None` we shouldn't set a custom icon to something other than `None`.

Reply:
What would you suggest for the device_class value if we don't want to change the custom icons?

Reply:
`None`

Reply:
Ok, I'll prepare a new PR with improvements including returning None as device_class.